gatsby-node.js

```diff
@@ -25,31 +25,32 @@ exports.createPages = async ({ actions, graphql }) => {
     query {
       allStrapiPage {
         nodes {
-          id: strapiId
-          slug
-          title
-          unlisted
-          strapiChildren {
-            id
-            slug
-            title
-          }
+          content
           strapiParent {
-            id
+            slug
           }
+          slug
+          title
           locale
-          content
         }
       }
     }
   `)
-
-  data.allStrapiPage.nodes.forEach((page) => {
+  const buildPathTree = (key, loc) => {
+    const node = data.allStrapiPage.nodes.find((obj) => obj.slug === key)
+    if (!node.strapiParent) {
+      return loc !== "fi" ? `${loc}/${node.slug}` : node.slug
+    }
+    return `${buildPathTree(node.strapiParent.slug, loc)}/${node.slug}`
+  }
```

Comment:
Maybe a small description about this function's logic could be useful as it's not super self-explanatory